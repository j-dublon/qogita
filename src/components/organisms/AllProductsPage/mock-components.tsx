import { AllProductsPageProps } from "@/types";
import React, { FC } from "react";

export const MockLoading: FC = () => <div>Loading...</div>;

export const MockError: FC = () => <div>Error</div>;

export const MockProductsList: FC = () => <div>Products List</div>;

export const MockPagination: FC = () => <div>Pagination</div>;

export const MockAllProductsPage: FC<AllProductsPageProps> = ({
  fetchData,
}) => (
  <div>
    Products Page
    <div onClick={() => fetchData(1)}>Products Page Pagination</div>
  </div>
);
