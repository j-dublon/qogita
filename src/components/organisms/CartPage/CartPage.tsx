import React, { FC } from "react";
import Layout from "../../Layout/index";
import { CartPageProps, ProductItem } from "@/types";
import Image from "next/image";
import CartQuantityButtons from "../../molecules/CartQuantityButtons/CartQuantityButtons";

const CartPage: FC<CartPageProps> = ({
  items,
  handleUpdateCart,
  cartTotal,
}) => (
  <Layout>
    <h1 className="text-3xl font-semibold text-center my-6">Your Cart</h1>
    {items.map((item) => (
      <div className="flex flex-row items-center justify-between bg-white p-4 rounded-lg shadow-lg mb-4">
        <div className="flex flex-row gap-4 items-center">
          <Image
            src={item.imageUrl}
            height={150}
            width={150}
            alt={`Image of ${item.name}`}
          />
          <h1 className="text-xl font-semibold mt-6 mb-8">{item.name}</h1>
        </div>

        <CartQuantityButtons
          numberInCart={item.quantity as number}
          updateCartQuantity={(newQuantity: number) => {
            handleUpdateCart(item as ProductItem, newQuantity);
          }}
        />
      </div>
    ))}
    <h1 className="text-3xl font-semibold text-center my-6">{`CART TOTAL: EUR ${cartTotal}`}</h1>
  </Layout>
);

export default CartPage;
