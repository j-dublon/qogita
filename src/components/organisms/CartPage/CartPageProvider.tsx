import React, { FC, useEffect, useState } from "react";
import CartPage from "./CartPage";
import { useCart } from "react-use-cart";
import { Loading } from "../../atoms";
import { ProductItem } from "@/types";

const CartPageProvider: FC = () => {
  const { items, updateItemQuantity, cartTotal, isEmpty } = useCart();
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    if ((items && cartTotal) || isEmpty) {
      setLoading(false);
    }
  }, [items, cartTotal, isEmpty]);

  const handleUpdateCart = (item: ProductItem, newQuantity: number) => {
    updateItemQuantity(item.id, newQuantity);
  };

  return (
    <>
      {loading ? (
        <Loading />
      ) : (
        <CartPage
          items={items}
          handleUpdateCart={handleUpdateCart}
          cartTotal={cartTotal}
        />
      )}
    </>
  );
};

export default CartPageProvider;
