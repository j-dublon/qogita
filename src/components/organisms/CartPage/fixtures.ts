import { CartPageProps } from "@/types";

export const mockProps: CartPageProps = {
  items: [
    {
      name: "Mock Product",
      imageUrl: "http://your-product.com",
      recommendedRetailPrice: 52,
      price: 52,
      recommendedRetailPriceCurrency: "EUR",
      gtin: "987",
      id: "987",
      brandName: "Chanel",
      categoryName: "Perfume",
      quantity: 3,
    },
    {
      name: "Mock Product Three",
      imageUrl: "http://your-product-three.com",
      recommendedRetailPrice: 12,
      price: 12,
      recommendedRetailPriceCurrency: "EUR",
      gtin: "123",
      id: "123",
      brandName: "Lynx",
      categoryName: "Deodorant",
      quantity: 2,
    },
  ],
  handleUpdateCart: jest.fn(),
  cartTotal: 356,
};
