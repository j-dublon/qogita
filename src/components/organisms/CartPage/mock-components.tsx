import { CartPageItemProps } from "@/types";
import React, { FC } from "react";

export const MockCartPageItem: FC<CartPageItemProps> = ({ item }) => (
  <div>{item.name}</div>
);
