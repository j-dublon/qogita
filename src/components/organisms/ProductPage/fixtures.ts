import { Product, ProductPageProps } from "@/types";
import { Item } from "react-use-cart";

export const mockAddItem = jest.fn();
export const mockUpdateItemQuantity = jest.fn();

export const mockProps: ProductPageProps = {
  product: {
    name: "Mock Product",
    imageUrl: "http://your-product.com",
    recommendedRetailPrice: 52,
    recommendedRetailPriceCurrency: "GBP",
    gtin: "987",
    brandName: "Chanel",
    categoryName: "Perfume",
  },
  numberInCart: 3,
  updateCartQuantity: jest.fn(),
};

export const mockCartItem = {
  ...mockProps.product,
  id: mockProps.product.gtin,
  price: mockProps.product.recommendedRetailPrice,
  quantity: 3,
};

export const alternativeProduct: Product = {
  name: "Other Mock Product",
  imageUrl: "http://your-product-two.com",
  recommendedRetailPrice: 35,
  recommendedRetailPriceCurrency: "GBP",
  gtin: "073",
  brandName: "Chanel",
  categoryName: "Perfume",
};

export const alternativeCartItem: Item = {
  name: "Other Mock Product",
  imageUrl: "http://your-product-two.com",
  recommendedRetailPrice: 35,
  price: 35,
  recommendedRetailPriceCurrency: "GBP",
  gtin: "073",
  id: "073",
  brandName: "Chanel",
  categoryName: "Perfume",
};
