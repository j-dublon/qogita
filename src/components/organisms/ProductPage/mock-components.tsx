import React, { FC } from "react";

export const MockUpdateCartQuantityButtons: FC = () => (
  <div>Update Cart Quantity Buttons</div>
);
