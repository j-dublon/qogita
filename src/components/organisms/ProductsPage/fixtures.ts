import { Product, ProductsPageProps } from "@/types";

const mockProducts: Product[] = [
  {
    name: "Parodontax Duplo Herbal Fresh 75ml",
    gtin: "5054563079435",
    recommendedRetailPrice: 29.99,
    recommendedRetailPriceCurrency: "EUR",
    imageUrl:
      "https://images.qogita.com/files/images/variants/aB9r5isuPDUTTD3nLNsXvQ.jpg",
    brandName: "Parodontax",
    categoryName: "Toothpaste",
  },
  {
    name: "Poseidon The Black Men Edt Vapo 150 Ml - Beauty & Health",
    gtin: "8411047151242",
    recommendedRetailPrice: 22.99,
    recommendedRetailPriceCurrency: "EUR",
    imageUrl:
      "https://images.qogita.com/files/images/variants/co8e7Y9gf272e2W2LgA6fj.jpg",
    brandName: "Instituto Espanol",
    categoryName: "Men's Perfume",
  },
  {
    name: "Snowracer Classic - Sled - Black",
    gtin: "7313327300382",
    recommendedRetailPrice: 90.49,
    recommendedRetailPriceCurrency: "EUR",
    imageUrl:
      "https://images.qogita.com/files/images/variants/jLSEtd5DRU72VXf7ScZ2m9.jpg",
    brandName: "Stiga",
    categoryName: "Sledding",
  },
];

export const mockProps: ProductsPageProps = {
  loading: false,
  error: false,
  products: mockProducts,
  currentPage: 1,
  totalPages: 5,
  fetchData: jest.fn(),
};

export const mockPropsLoading: ProductsPageProps = {
  ...mockProps,
  loading: true,
};

export const mockPropsErrorState: ProductsPageProps = {
  ...mockProps,
  error: true,
};
