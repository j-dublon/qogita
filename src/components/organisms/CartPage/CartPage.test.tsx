import React from "react";
import { render } from "@testing-library/react";
import CartPage from "./CartPage";
import { mockProps } from "./fixtures";
import { MockCartPageItem } from "./mock-components";
import { CartPageItemProps } from "@/types";

jest.mock("../../molecules/CartPageItem/CartPageItem", () => ({
  __esModule: true,
  default: (props: CartPageItemProps) => <MockCartPageItem {...props} />,
}));

describe("Component: CartPage", () => {
  it("SHOULD match snapshot", () => {
    const component = render(<CartPage {...mockProps} />);
    expect(component).toMatchSnapshot();
  });

  it("SHOULD render page title and cart total as expected WHEN props are provided", () => {
    const { getByText, getAllByText } = render(<CartPage {...mockProps} />);
    expect(getAllByText("Your Cart")).toHaveLength(2);
    expect(getByText(`CART TOTAL: EUR ${mockProps.cartTotal}`));
  });

  it("SHOULD render a CartPageItem component for each item WHEN props are provided", () => {
    const { getByText } = render(<CartPage {...mockProps} />);
    mockProps.items.forEach((item) => expect(getByText(item.name)));
  });
});
