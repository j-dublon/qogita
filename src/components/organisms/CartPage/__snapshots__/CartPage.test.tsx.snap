// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: CartPage SHOULD match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container mx-auto px-4"
      >
        <div
          class="flex justify-between"
        >
          <strong>
            Qogita
          </strong>
          <nav>
            <ul
              class="flex gap-4"
            >
              <li>
                <a
                  class="underline"
                  href="/"
                >
                  Products
                </a>
              </li>
              <li
                class="flex flex-row"
              >
                <a
                  class="underline mr-1"
                  href="/cart"
                >
                  Your Cart
                </a>
                <p />
              </li>
            </ul>
          </nav>
        </div>
        <h1
          class="text-3xl font-semibold text-center my-6"
        >
          Your Cart
        </h1>
        <div>
          Mock Product
        </div>
        <div>
          Mock Product Three
        </div>
        <h1
          class="text-3xl font-semibold text-center my-6"
        >
          CART TOTAL: EUR 356.00
        </h1>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container mx-auto px-4"
    >
      <div
        class="flex justify-between"
      >
        <strong>
          Qogita
        </strong>
        <nav>
          <ul
            class="flex gap-4"
          >
            <li>
              <a
                class="underline"
                href="/"
              >
                Products
              </a>
            </li>
            <li
              class="flex flex-row"
            >
              <a
                class="underline mr-1"
                href="/cart"
              >
                Your Cart
              </a>
              <p />
            </li>
          </ul>
        </nav>
      </div>
      <h1
        class="text-3xl font-semibold text-center my-6"
      >
        Your Cart
      </h1>
      <div>
        Mock Product
      </div>
      <div>
        Mock Product Three
      </div>
      <h1
        class="text-3xl font-semibold text-center my-6"
      >
        CART TOTAL: EUR 356.00
      </h1>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
